backend/app/services/subscription/execution.py

```
@@ -235,6 +235,23 @@ def cancel_execution(
+        if execution.task_id and execution.task_id > 0:
+            # Run async cancel in background to not block the response
+            # This ensures executor is notified to stop the actual execution
+            import asyncio
+
+            try:
+                # Try to get running loop (Python 3.10+ recommended way)
+                loop = asyncio.get_running_loop()
+                # Schedule as background task
+                asyncio.create_task(
+                    self.cancel_task_by_id(db, execution.task_id, user_id)
+                )
```

Ruff (0.15.9)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)